class08
================
Kendall Lin
1/30/2020

Guest Lecture: Notes: - Nucleosomes come together to form chromatin
fiber - Chromatin is nucleosomes with DNA wrapped around it - Nuclear
pores - can get genes, RNA - Nucleolus is mapped with fractionation -
Epigenetic marks demarcate the start and end of genes and provide
structure to the chromosome-\> alter how we read them - Methylation adds
methyl group to cytosine group on DNA - Methylation is inherited by
mitotic division -\> enzymes like DNMT1 recognize hemi-methylated DNA(1
strand methylated) - DNA Demethylation -\> remove methyl group on
cytosine - Active demethylation and Passive demethylation exists -
hypomethylation of repeat elements and hypermethylation of CpG islands
can lead to cancer

Detecting Methylation -\> Bisulfite conversion -\> if DNA is
unmethylated, the cytosine turns into uracil and when sequenced it
becomes Tyrosine -\> compare the reference genome to the bisulfite one
to see sites of methylation

\-4 Kinds of histone proteins - Gene suppression = high nucleosome
density/high repressive methylation and hypoacetylation - Gene
activation = opposite -Lysine(K) tails of H3 and H4 are very important
for modification - H3K27me3 -\> histone 3, lysine 27, 3 methyls attached

Histone Acetylation - HATs and HDACs - reduces positive charge of
histones-\> less attraction b/w histone and -ve charged DNA loosen
histones - Acetylated histones are docking sites for other proteins that
opens chromatin further - No clear inheritance mechanism unlike DNA
methylation

Histone Methylation - Methyl groups are laid onto lysines/K (different
from DNA methyl which targets Cytosine) - can be mono/di/tri (1,2,3) -
Does not change electrical charge of histones - Can positively or
negatively affect gene expression -H3K9me -\> inactive ; H3k27me -\>
inactive

\-euchromatin and heterochromatin

ChIP Seq -\> use immunoprecipitate
